### ISSUE TITLE
collaborative mode doesn't save or sync cell output

### ISSUE THREAD

**Issue by minrk** [jupyterlab/jupyterlab#13182]
## Description

Running with `jupyter lab --collaborative` (4.0a29), cell outputs are not synced or saved.

Screenshot immediately after execute on the right hand side: input and cursors remain properly synced, but outputs never show up in clients that didn't produce them:

<img width="1742" alt="Screen Shot 2022-10-04 at 17 57 15" src="https://user-images.githubusercontent.com/151929/193956887-21040f03-3ad9-411a-a235-e9401ed769b6.png">

Refreshing the page which produced the output (right) loses the output (matches left):

<img width="1737" alt="Screen Shot 2022-10-04 at 18 07 54" src="https://user-images.githubusercontent.com/151929/193958118-c505d2dd-c742-4243-8f06-d5fa2ac7a94a.png">

Both sides can produce output, and neither side produces output that is persisted either to disk or to peers:

<img width="1737" alt="Screen Shot 2022-10-04 at 18 08 14" src="https://user-images.githubusercontent.com/151929/193957999-49bdf054-a963-440b-8ad4-5d53b1accb11.png">

## Reproduce

1. launch `jupyter lab --collaborative`
2. launch a second browser at the same URL
3. open a notebook and execute a cell that produces output
4. input is synced, output never shows up where execution didn't happen
5. refresh page that produced output and output will be lost (matches peer)

Testing RTC, noticed output is not synced. Reproduced in clean env, with only `pip install --upgrade --pre jupyterlab`

## Expected behavior

output is included in the document model that is synced and saved.

## Context

- Operating System and version: macOS
- Browser and version: Safari 16, Firefox 
- JupyterLab version: 4.0.0a29

<details><summary>Troubleshoot Output</summary>
<pre>
$PATH:
	/opt/homebrew/opt/ruby/bin
	/Users/minrk/conda/envs/jlab-test/bin
	/usr/local/opt/ruby/bin
	/Users/minrk/dev/mine/git-stuff/bin
	/Users/minrk/conda/condabin
	/opt/homebrew/Caskroom/google-cloud-sdk/latest/google-cloud-sdk/bin
	/Users/minrk/.local/bin
	/Users/minrk/conda/bin
	/usr/local/texlive/2022/bin/universal-darwin
	/Users/minrk/Dropbox/bin
	/opt/homebrew/bin
	/usr/local/MacGPG2/bin
	/usr/local/Homebrew/bin
	/usr/local/bin
	/usr/X11R6/bin
	/usr/X11/bin
	/usr/bin
	/opt/pypy/bin
	/Users/minrk/Dropbox/scripts
	/bin
	/usr/sbin
	/sbin
	/Library/TeX/texbin
	/opt/X11/bin
	/Library/Apple/usr/bin
	/Library/Frameworks/Mono.framework/Versions/Current/Commands

sys.path:
	/Users/minrk/conda/envs/jlab-test/bin
	/Users/minrk/conda/envs/jlab-test/lib/python310.zip
	/Users/minrk/conda/envs/jlab-test/lib/python3.10
	/Users/minrk/conda/envs/jlab-test/lib/python3.10/lib-dynload
	/Users/minrk/conda/envs/jlab-test/lib/python3.10/site-packages

sys.executable:
	/Users/minrk/conda/envs/jlab-test/bin/python

sys.version:
	3.10.6 | packaged by conda-forge | (main, Aug 22 2022, 20:41:22) [Clang 13.0.1 ]

platform.platform():
	macOS-12.6-arm64-arm-64bit

which -a jupyter:
	/Users/minrk/conda/envs/jlab-test/bin/jupyter
	/Users/minrk/conda/bin/jupyter

pip list:
	Package                       Version
	----------------------------- -----------
	aiofiles                      0.8.0
	aiosqlite                     0.17.0
	anyio                         3.6.1
	appnope                       0.1.3
	argon2-cffi                   21.3.0
	argon2-cffi-bindings          21.2.0
	arrow                         1.2.3
	asttokens                     2.0.8
	async-lru                     1.0.3
	attrs                         22.1.0
	Babel                         2.10.3
	backcall                      0.2.0
	backports.functools-lru-cache 1.6.4
	beautifulsoup4                4.11.1
	bleach                        5.0.1
	brotlipy                      0.7.0
	certifi                       2022.9.24
	cffi                          1.15.1
	charset-normalizer            2.1.1
	cryptography                  38.0.1
	debugpy                       1.6.3
	decorator                     5.1.1
	defusedxml                    0.7.1
	entrypoints                   0.4
	executing                     1.1.0
	fastjsonschema                2.16.2
	flit_core                     3.7.1
	fqdn                          1.5.1
	idna                          3.4
	importlib-metadata            4.11.4
	importlib-resources           5.9.0
	ipykernel                     6.16.0
	ipython                       8.5.0
	ipython-genutils              0.2.0
	isoduration                   20.11.0
	jedi                          0.18.1
	Jinja2                        3.1.2
	json5                         0.9.5
	jsonpointer                   2.3
	jsonschema                    4.16.0
	jupyter_client                7.3.5
	jupyter_core                  4.11.1
	jupyter-events                0.5.0
	jupyter-lsp                   1.5.1
	jupyter-server                1.19.1
	jupyter_server_terminals      0.3.1
	jupyter_server_ydoc           0.1.12
	jupyter-ydoc                  0.2.1
	jupyterlab                    4.0.0a29
	jupyterlab-pygments           0.2.2
	jupyterlab_server             2.15.2
	MarkupSafe                    2.1.1
	matplotlib-inline             0.1.6
	mistune                       0.8.4
	nbclassic                     0.3.7
	nbclient                      0.5.13
	nbconvert                     6.4.5
	nbformat                      5.6.1
	nest-asyncio                  1.5.6
	notebook                      6.4.12
	notebook-shim                 0.1.0
	packaging                     21.3
	pandocfilters                 1.5.0
	parso                         0.8.3
	pexpect                       4.8.0
	pickleshare                   0.7.5
	pip                           22.2.2
	pkgutil_resolve_name          1.3.10
	prometheus-client             0.14.1
	prompt-toolkit                3.0.31
	psutil                        5.9.2
	ptyprocess                    0.7.0
	pure-eval                     0.2.2
	pycparser                     2.21
	Pygments                      2.13.0
	pyOpenSSL                     22.0.0
	pyparsing                     3.0.9
	pyrsistent                    0.18.1
	PySocks                       1.7.1
	python-dateutil               2.8.2
	python-json-logger            2.0.4
	pytz                          2022.4
	PyYAML                        6.0
	pyzmq                         24.0.1
	requests                      2.28.1
	rfc3339-validator             0.1.4
	rfc3986-validator             0.1.1
	Send2Trash                    1.8.0
	setuptools                    65.4.1
	six                           1.16.0
	sniffio                       1.3.0
	soupsieve                     2.3.2.post1
	stack-data                    0.5.1
	terminado                     0.16.0
	testpath                      0.6.0
	tomli                         2.0.1
	tornado                       6.2
	traitlets                     5.4.0
	typing_extensions             4.3.0
	uri-template                  1.2.0
	urllib3                       1.26.11
	wcwidth                       0.2.5
	webcolors                     1.12
	webencodings                  0.5.1
	websocket-client              1.4.1
	wheel                         0.37.1
	y-py                          0.5.4
	ypy-websocket                 0.3.2
	zipp                          3.8.1

conda list:
	# packages in environment at /Users/minrk/conda/envs/jlab-test:
	#
	# Name                    Version                   Build  Channel
	aiofiles                  0.8.0                    pypi_0    pypi
	aiosqlite                 0.17.0                   pypi_0    pypi
	anyio                     3.6.1              pyhd8ed1ab_1    conda-forge
	appnope                   0.1.3              pyhd8ed1ab_0    conda-forge
	argon2-cffi               21.3.0             pyhd8ed1ab_0    conda-forge
	argon2-cffi-bindings      21.2.0          py310hf8d0d8f_2    conda-forge
	arrow                     1.2.3                    pypi_0    pypi
	asttokens                 2.0.8              pyhd8ed1ab_0    conda-forge
	async-lru                 1.0.3                    pypi_0    pypi
	attrs                     22.1.0             pyh71513ae_1    conda-forge
	babel                     2.10.3             pyhd8ed1ab_0    conda-forge
	backcall                  0.2.0              pyh9f0ad1d_0    conda-forge
	backports                 1.0                        py_2    conda-forge
	backports.functools_lru_cache 1.6.4              pyhd8ed1ab_0    conda-forge
	beautifulsoup4            4.11.1             pyha770c72_0    conda-forge
	bleach                    5.0.1              pyhd8ed1ab_0    conda-forge
	brotlipy                  0.7.0           py310hf8d0d8f_1004    conda-forge
	bzip2                     1.0.8                h3422bc3_4    conda-forge
	ca-certificates           2022.9.24            h4653dfc_0    conda-forge
	certifi                   2022.9.24          pyhd8ed1ab_0    conda-forge
	cffi                      1.15.1          py310he00a5c5_0    conda-forge
	charset-normalizer        2.1.1              pyhd8ed1ab_0    conda-forge
	cryptography              38.0.1          py310h4fe9c50_0    conda-forge
	debugpy                   1.6.3           py310hc6dc59f_0    conda-forge
	decorator                 5.1.1              pyhd8ed1ab_0    conda-forge
	defusedxml                0.7.1              pyhd8ed1ab_0    conda-forge
	entrypoints               0.4                pyhd8ed1ab_0    conda-forge
	executing                 1.1.0              pyhd8ed1ab_0    conda-forge
	flit-core                 3.7.1              pyhd8ed1ab_0    conda-forge
	fqdn                      1.5.1                    pypi_0    pypi
	idna                      3.4                pyhd8ed1ab_0    conda-forge
	importlib-metadata        4.11.4          py310hbe9552e_0    conda-forge
	importlib_resources       5.9.0              pyhd8ed1ab_0    conda-forge
	ipykernel                 6.16.0             pyh736e0ef_0    conda-forge
	ipython                   8.5.0              pyhd1c38e8_1    conda-forge
	ipython_genutils          0.2.0                      py_1    conda-forge
	isoduration               20.11.0                  pypi_0    pypi
	jedi                      0.18.1             pyhd8ed1ab_2    conda-forge
	jinja2                    3.1.2              pyhd8ed1ab_1    conda-forge
	json5                     0.9.5              pyh9f0ad1d_0    conda-forge
	jsonpointer               2.3                      pypi_0    pypi
	jsonschema                4.16.0             pyhd8ed1ab_0    conda-forge
	jupyter-events            0.5.0                    pypi_0    pypi
	jupyter-lsp               1.5.1                    pypi_0    pypi
	jupyter-server            1.19.1                   pypi_0    pypi
	jupyter-server-terminals  0.3.1                    pypi_0    pypi
	jupyter-server-ydoc       0.1.12                   pypi_0    pypi
	jupyter-ydoc              0.2.1                    pypi_0    pypi
	jupyter_client            7.3.5              pyhd8ed1ab_0    conda-forge
	jupyter_core              4.11.1          py310hbe9552e_0    conda-forge
	jupyterlab                4.0.0a29                 pypi_0    pypi
	jupyterlab_pygments       0.2.2              pyhd8ed1ab_0    conda-forge
	jupyterlab_server         2.15.2             pyhd8ed1ab_0    conda-forge
	libcxx                    14.0.6               h2692d47_0    conda-forge
	libffi                    3.4.2                h3422bc3_5    conda-forge
	libsodium                 1.0.18               h27ca646_1    conda-forge
	libsqlite                 3.39.4               h76d750c_0    conda-forge
	libzlib                   1.2.12               h03a7124_4    conda-forge
	markupsafe                2.1.1           py310hf8d0d8f_1    conda-forge
	matplotlib-inline         0.1.6              pyhd8ed1ab_0    conda-forge
	mistune                   0.8.4           py310he2143c4_1005    conda-forge
	nbclassic                 0.3.7              pyhd8ed1ab_0    conda-forge
	nbclient                  0.5.13             pyhd8ed1ab_0    conda-forge
	nbconvert                 6.4.5           py310hbe9552e_0    conda-forge
	nbformat                  5.6.1              pyhd8ed1ab_0    conda-forge
	ncurses                   6.3                  h07bb92c_1    conda-forge
	nest-asyncio              1.5.6              pyhd8ed1ab_0    conda-forge
	notebook                  6.4.12             pyha770c72_0    conda-forge
	notebook-shim             0.1.0              pyhd8ed1ab_0    conda-forge
	openssl                   1.1.1q               ha287fd2_0    conda-forge
	packaging                 21.3               pyhd8ed1ab_0    conda-forge
	pandocfilters             1.5.0              pyhd8ed1ab_0    conda-forge
	parso                     0.8.3              pyhd8ed1ab_0    conda-forge
	pexpect                   4.8.0              pyh9f0ad1d_2    conda-forge
	pickleshare               0.7.5                   py_1003    conda-forge
	pip                       22.2.2             pyhd8ed1ab_0    conda-forge
	pkgutil-resolve-name      1.3.10             pyhd8ed1ab_0    conda-forge
	prometheus_client         0.14.1             pyhd8ed1ab_0    conda-forge
	prompt-toolkit            3.0.31             pyha770c72_0    conda-forge
	psutil                    5.9.2           py310h8e9501a_0    conda-forge
	ptyprocess                0.7.0              pyhd3deb0d_0    conda-forge
	pure_eval                 0.2.2              pyhd8ed1ab_0    conda-forge
	pycparser                 2.21               pyhd8ed1ab_0    conda-forge
	pygments                  2.13.0             pyhd8ed1ab_0    conda-forge
	pyopenssl                 22.0.0             pyhd8ed1ab_1    conda-forge
	pyparsing                 3.0.9              pyhd8ed1ab_0    conda-forge
	pyrsistent                0.18.1          py310hf8d0d8f_1    conda-forge
	pysocks                   1.7.1              pyha2e5f31_6    conda-forge
	python                    3.10.6          hbce4517_0_cpython    conda-forge
	python-dateutil           2.8.2              pyhd8ed1ab_0    conda-forge
	python-fastjsonschema     2.16.2             pyhd8ed1ab_0    conda-forge
	python-json-logger        2.0.4                    pypi_0    pypi
	python_abi                3.10                    2_cp310    conda-forge
	pytz                      2022.4             pyhd8ed1ab_0    conda-forge
	pyyaml                    6.0                      pypi_0    pypi
	pyzmq                     24.0.1          py310hc407298_0    conda-forge
	readline                  8.1.2                h46ed386_0    conda-forge
	requests                  2.28.1             pyhd8ed1ab_1    conda-forge
	rfc3339-validator         0.1.4                    pypi_0    pypi
	rfc3986-validator         0.1.1                    pypi_0    pypi
	send2trash                1.8.0              pyhd8ed1ab_0    conda-forge
	setuptools                65.4.1             pyhd8ed1ab_0    conda-forge
	six                       1.16.0             pyh6c4a22f_0    conda-forge
	sniffio                   1.3.0              pyhd8ed1ab_0    conda-forge
	soupsieve                 2.3.2.post1        pyhd8ed1ab_0    conda-forge
	stack_data                0.5.1              pyhd8ed1ab_0    conda-forge
	terminado                 0.16.0             pyhd1c38e8_0    conda-forge
	testpath                  0.6.0              pyhd8ed1ab_0    conda-forge
	tk                        8.6.12               he1e0b03_0    conda-forge
	tomli                     2.0.1              pyhd8ed1ab_0    conda-forge
	tornado                   6.2             py310h02f21da_0    conda-forge
	traitlets                 5.4.0              pyhd8ed1ab_0    conda-forge
	typing_extensions         4.3.0              pyha770c72_0    conda-forge
	tzdata                    2022d                h191b570_0    conda-forge
	uri-template              1.2.0                    pypi_0    pypi
	urllib3                   1.26.11            pyhd8ed1ab_0    conda-forge
	wcwidth                   0.2.5              pyh9f0ad1d_2    conda-forge
	webcolors                 1.12                     pypi_0    pypi
	webencodings              0.5.1                      py_1    conda-forge
	websocket-client          1.4.1              pyhd8ed1ab_0    conda-forge
	wheel                     0.37.1             pyhd8ed1ab_0    conda-forge
	xz                        5.2.6                h57fd34a_0    conda-forge
	y-py                      0.5.4                    pypi_0    pypi
	ypy-websocket             0.3.2                    pypi_0    pypi
	zeromq                    4.3.4                hbdafb3b_1    conda-forge
	zipp                      3.8.1              pyhd8ed1ab_0    conda-forge

conda env:
	name: jlab-test
	channels:
	  - conda-forge
	  - defaults
	dependencies:
	  - anyio=3.6.1=pyhd8ed1ab_1
	  - appnope=0.1.3=pyhd8ed1ab_0
	  - argon2-cffi=21.3.0=pyhd8ed1ab_0
	  - argon2-cffi-bindings=21.2.0=py310hf8d0d8f_2
	  - asttokens=2.0.8=pyhd8ed1ab_0
	  - attrs=22.1.0=pyh71513ae_1
	  - babel=2.10.3=pyhd8ed1ab_0
	  - backcall=0.2.0=pyh9f0ad1d_0
	  - backports=1.0=py_2
	  - backports.functools_lru_cache=1.6.4=pyhd8ed1ab_0
	  - beautifulsoup4=4.11.1=pyha770c72_0
	  - bleach=5.0.1=pyhd8ed1ab_0
	  - brotlipy=0.7.0=py310hf8d0d8f_1004
	  - bzip2=1.0.8=h3422bc3_4
	  - ca-certificates=2022.9.24=h4653dfc_0
	  - certifi=2022.9.24=pyhd8ed1ab_0
	  - cffi=1.15.1=py310he00a5c5_0
	  - charset-normalizer=2.1.1=pyhd8ed1ab_0
	  - cryptography=38.0.1=py310h4fe9c50_0
	  - debugpy=1.6.3=py310hc6dc59f_0
	  - decorator=5.1.1=pyhd8ed1ab_0
	  - defusedxml=0.7.1=pyhd8ed1ab_0
	  - entrypoints=0.4=pyhd8ed1ab_0
	  - executing=1.1.0=pyhd8ed1ab_0
	  - flit-core=3.7.1=pyhd8ed1ab_0
	  - idna=3.4=pyhd8ed1ab_0
	  - importlib-metadata=4.11.4=py310hbe9552e_0
	  - importlib_resources=5.9.0=pyhd8ed1ab_0
	  - ipykernel=6.16.0=pyh736e0ef_0
	  - ipython=8.5.0=pyhd1c38e8_1
	  - ipython_genutils=0.2.0=py_1
	  - jedi=0.18.1=pyhd8ed1ab_2
	  - jinja2=3.1.2=pyhd8ed1ab_1
	  - json5=0.9.5=pyh9f0ad1d_0
	  - jsonschema=4.16.0=pyhd8ed1ab_0
	  - jupyter_client=7.3.5=pyhd8ed1ab_0
	  - jupyter_core=4.11.1=py310hbe9552e_0
	  - jupyterlab_pygments=0.2.2=pyhd8ed1ab_0
	  - jupyterlab_server=2.15.2=pyhd8ed1ab_0
	  - libcxx=14.0.6=h2692d47_0
	  - libffi=3.4.2=h3422bc3_5
	  - libsodium=1.0.18=h27ca646_1
	  - libsqlite=3.39.4=h76d750c_0
	  - libzlib=1.2.12=h03a7124_4
	  - markupsafe=2.1.1=py310hf8d0d8f_1
	  - matplotlib-inline=0.1.6=pyhd8ed1ab_0
	  - mistune=0.8.4=py310he2143c4_1005
	  - nbclassic=0.3.7=pyhd8ed1ab_0
	  - nbclient=0.5.13=pyhd8ed1ab_0
	  - nbconvert=6.4.5=py310hbe9552e_0
	  - nbformat=5.6.1=pyhd8ed1ab_0
	  - ncurses=6.3=h07bb92c_1
	  - nest-asyncio=1.5.6=pyhd8ed1ab_0
	  - notebook=6.4.12=pyha770c72_0
	  - notebook-shim=0.1.0=pyhd8ed1ab_0
	  - openssl=1.1.1q=ha287fd2_0
	  - packaging=21.3=pyhd8ed1ab_0
	  - pandocfilters=1.5.0=pyhd8ed1ab_0
	  - parso=0.8.3=pyhd8ed1ab_0
	  - pexpect=4.8.0=pyh9f0ad1d_2
	  - pickleshare=0.7.5=py_1003
	  - pip=22.2.2=pyhd8ed1ab_0
	  - pkgutil-resolve-name=1.3.10=pyhd8ed1ab_0
	  - prometheus_client=0.14.1=pyhd8ed1ab_0
	  - prompt-toolkit=3.0.31=pyha770c72_0
	  - psutil=5.9.2=py310h8e9501a_0
	  - ptyprocess=0.7.0=pyhd3deb0d_0
	  - pure_eval=0.2.2=pyhd8ed1ab_0
	  - pycparser=2.21=pyhd8ed1ab_0
	  - pygments=2.13.0=pyhd8ed1ab_0
	  - pyopenssl=22.0.0=pyhd8ed1ab_1
	  - pyparsing=3.0.9=pyhd8ed1ab_0
	  - pyrsistent=0.18.1=py310hf8d0d8f_1
	  - pysocks=1.7.1=pyha2e5f31_6
	  - python=3.10.6=hbce4517_0_cpython
	  - python-dateutil=2.8.2=pyhd8ed1ab_0
	  - python-fastjsonschema=2.16.2=pyhd8ed1ab_0
	  - python_abi=3.10=2_cp310
	  - pytz=2022.4=pyhd8ed1ab_0
	  - pyzmq=24.0.1=py310hc407298_0
	  - readline=8.1.2=h46ed386_0
	  - requests=2.28.1=pyhd8ed1ab_1
	  - send2trash=1.8.0=pyhd8ed1ab_0
	  - setuptools=65.4.1=pyhd8ed1ab_0
	  - six=1.16.0=pyh6c4a22f_0
	  - sniffio=1.3.0=pyhd8ed1ab_0
	  - soupsieve=2.3.2.post1=pyhd8ed1ab_0
	  - stack_data=0.5.1=pyhd8ed1ab_0
	  - terminado=0.16.0=pyhd1c38e8_0
	  - testpath=0.6.0=pyhd8ed1ab_0
	  - tk=8.6.12=he1e0b03_0
	  - tomli=2.0.1=pyhd8ed1ab_0
	  - tornado=6.2=py310h02f21da_0
	  - traitlets=5.4.0=pyhd8ed1ab_0
	  - typing_extensions=4.3.0=pyha770c72_0
	  - tzdata=2022d=h191b570_0
	  - urllib3=1.26.11=pyhd8ed1ab_0
	  - wcwidth=0.2.5=pyh9f0ad1d_2
	  - webencodings=0.5.1=py_1
	  - websocket-client=1.4.1=pyhd8ed1ab_0
	  - wheel=0.37.1=pyhd8ed1ab_0
	  - xz=5.2.6=h57fd34a_0
	  - zeromq=4.3.4=hbdafb3b_1
	  - zipp=3.8.1=pyhd8ed1ab_0
	  - pip:
	    - aiofiles==0.8.0
	    - aiosqlite==0.17.0
	    - arrow==1.2.3
	    - async-lru==1.0.3
	    - fqdn==1.5.1
	    - isoduration==20.11.0
	    - jsonpointer==2.3
	    - jupyter-events==0.5.0
	    - jupyter-lsp==1.5.1
	    - jupyter-server==1.19.1
	    - jupyter-server-terminals==0.3.1
	    - jupyter-server-ydoc==0.1.12
	    - jupyter-ydoc==0.2.1
	    - jupyterlab==4.0.0a29
	    - python-json-logger==2.0.4
	    - pyyaml==6.0
	    - rfc3339-validator==0.1.4
	    - rfc3986-validator==0.1.1
	    - uri-template==1.2.0
	    - webcolors==1.12
	    - y-py==0.5.4
	    - ypy-websocket==0.3.2
	prefix: /Users/minrk/conda/envs/jlab-test
</pre>
</details>

<details><summary>Command Line Output</summary>
<pre>
$ jupyter lab --collaborative
[I 2022-10-04 17:54:55.838 ServerApp] jupyter_lsp | extension was successfully linked.
[I 2022-10-04 17:54:55.841 ServerApp] jupyter_server_terminals | extension was successfully linked.
[I 2022-10-04 17:54:55.841 ServerApp] jupyter_server_ydoc | extension was successfully linked.
[I 2022-10-04 17:54:55.848 ServerApp] jupyterlab | extension was successfully linked.
[W 2022-10-04 17:54:55.850 NotebookApp] 'notebook_dir' has moved from NotebookApp to ServerApp. This config will be passed to ServerApp. Be sure to update your config before our next release.
[W 2022-10-04 17:54:55.850 NotebookApp] 'password' has moved from NotebookApp to ServerApp. This config will be passed to ServerApp. Be sure to update your config before our next release.
[W 2022-10-04 17:54:55.850 NotebookApp] 'password' has moved from NotebookApp to ServerApp. This config will be passed to ServerApp. Be sure to update your config before our next release.
[I 2022-10-04 17:54:55.854 ServerApp] nbclassic | extension was successfully linked.
[I 2022-10-04 17:54:55.959 ServerApp] notebook_shim | extension was successfully linked.
[I 2022-10-04 17:54:55.975 ServerApp] notebook_shim | extension was successfully loaded.
[I 2022-10-04 17:54:57.297 ServerApp] Skipped non-installed server(s): bash-language-server, dockerfile-language-server-nodejs, javascript-typescript-langserver, jedi-language-server, julia-language-server, pyright, python-language-server, python-lsp-server, r-languageserver, sql-language-server, typescript-language-server, unified-language-server, vscode-css-languageserver-bin, vscode-html-languageserver-bin, vscode-json-languageserver-bin, yaml-language-server
[I 2022-10-04 17:54:57.321 ServerApp] jupyter_lsp | extension was successfully loaded.
[I 2022-10-04 17:54:57.322 ServerApp] jupyter_server_terminals | extension was successfully loaded.
[I 2022-10-04 17:54:57.322 ServerApp] jupyter_server_ydoc | extension was successfully loaded.
[I 2022-10-04 17:54:57.323 LabApp] JupyterLab extension loaded from /Users/minrk/conda/envs/jlab-test/lib/python3.10/site-packages/jupyterlab
[I 2022-10-04 17:54:57.323 LabApp] JupyterLab application directory is /Users/minrk/conda/envs/jlab-test/share/jupyter/lab
[I 2022-10-04 17:54:57.323 LabApp] Extension Manager is 'pypi'.
[I 2022-10-04 17:54:57.325 ServerApp] jupyterlab | extension was successfully loaded.
[I 2022-10-04 17:54:57.328 ServerApp] nbclassic | extension was successfully loaded.
[I 2022-10-04 17:54:57.329 ServerApp] Serving notebooks from local directory: /Users/minrk/dev/mine/notebooks
[I 2022-10-04 17:54:57.329 ServerApp] Jupyter Server 1.19.1 is running at:
[I 2022-10-04 17:54:57.329 ServerApp] http://localhost:8888/lab
[I 2022-10-04 17:54:57.329 ServerApp]  or http://127.0.0.1:8888/lab
[I 2022-10-04 17:54:57.329 ServerApp] Use Control-C to stop this server and shut down all kernels (twice to skip confirmation).
[I 2022-10-04 17:55:04.908 ServerApp] 302 POST /login?next=%2Flab%2Ftree%2FUntitled.ipynb (::1) 171.87ms
[I 2022-10-04 17:55:06.405 LabApp] Build is up to date
[W 2022-10-04 17:55:06.522 ServerApp] Notebook Untitled11.ipynb is not trusted
[I 2022-10-04 17:55:06.739 ServerApp] Kernel started: 4d18cfc0-9263-4f9e-a2e1-9e07c6f11804</pre>
</details>

**Comment by hbcarlos** [comment]
Thanks, @minrk!
Indeed, the outputs are not shared. It was probably introduced in https://github.com/jupyterlab/jupyterlab/pull/12695.
I'll take a look.

**Comment by isms** [comment]
Not sure if this is helpful, but when paired with `--dev-mode` I also see the following two JS errors in the developer console:

```
Starting application in workspace: "default" index.tsx:516:12
Starting WebSocket: ws://localhost:8888/api/kernels/5055f9a4-01e1-4094-854a-feca9a371995 default.ts:1254:12
InternalError: too much recursion
    deepObjectCopy index.js:179
    deepCopy index.js:101
    deepObjectCopy index.js:187
    deepCopy index.js:101
    deepObjectCopy index.js:187
    deepCopy index.js:101
    deepObjectCopy index.js:187
    deepCopy index.js:101
    deepObjectCopy index.js:187
    deepCopy index.js:101
    deepObjectCopy index.js:187
index.es6.js:210
    exceptionHandler index.es6.js:210
    invokeSlot index.es6.js:460
    emit index.es6.js:415
    emit index.es6.js:99
    _onYCellsChanged ymodels.ts:611
    callAll function.js:19
    callEventHandlerListeners yjs.mjs:1885
    callTypeObservers yjs.mjs:4642
    _callObserver yjs.mjs:5401
    cleanupTransactions yjs.mjs:3097
    callAll function.js:19
    cleanupTransactions yjs.mjs:3127
    transact yjs.mjs:3239
    readUpdateV2 yjs.mjs:1550
    applyUpdateV2 yjs.mjs:1645
    applyUpdate yjs.mjs:1659
    readSyncStep2 sync.js:83
    readSyncMessage sync.js:121
    messageSync y-websocket.js:36
    readMessage y-websocket.js:76
    onmessage y-websocket.js:97
    (Async: EventHandlerNonNull)
    setupWS y-websocket.js:95
    connect y-websocket.js:393
    WebsocketProvider y-websocket.js:305
    WebSocketProvider yprovider.ts:33
    factory index.ts:41
    Context context.ts:69
    _createContext manager.ts:508
    _createOrOpenDocument manager.ts:602
    open manager.ts:378
    openOrReveal manager.ts:417
    execute index.tsx:629
    (Async: promise callback)
    execute index.tsx:629
    execute index.es6.js:352
    values restorablepool.ts:265
    restore restorablepool.ts:255
    restore widgettracker.ts:321
    restore layoutrestorer.ts:259
    activateNotebookHandler index.ts:1505
    promise index.es6.js:1827
    (Async: promise callback)
    activatePlugin index.es6.js:1827
    resolveRequiredService index.es6.js:1914
    required index.es6.js:1822
    activatePlugin index.es6.js:1822
    resolveRequiredService index.es6.js:1914
    required index.es6.js:1822
    activatePlugin index.es6.js:1822
    promises index.es6.js:1990
    start index.es6.js:1989
    main index.out.js:626
    bootstrap bootstrap.js:96
    js bootstrap.js:97
    Webpack 5
```

and:

```
TypeError: cell is undefined
    _insertCell widget.ts:623
    _onCellsChanged widget.ts:564
    _onCellsChanged widget.ts:563
    _onCellsChanged widget.ts:557
    _onCellsChanged widget.ts:1226
    invokeSlot index.es6.js:457
    emit index.es6.js:415
    emit index.es6.js:99
    _onYCellsChanged ymodels.ts:611
    callAll function.js:19
    callEventHandlerListeners yjs.mjs:1885
    callTypeObservers yjs.mjs:4642
    _callObserver yjs.mjs:5401
    cleanupTransactions yjs.mjs:3097
    callAll function.js:19
    cleanupTransactions yjs.mjs:3127
    transact yjs.mjs:3239
    readUpdateV2 yjs.mjs:1550
    applyUpdateV2 yjs.mjs:1645
    applyUpdate yjs.mjs:1659
    readSyncStep2 sync.js:83
    readSyncMessage sync.js:121
    messageSync y-websocket.js:36
    readMessage y-websocket.js:76
    onmessage y-websocket.js:97
    setupWS y-websocket.js:95
    connect y-websocket.js:393
    WebsocketProvider y-websocket.js:305
    WebSocketProvider yprovider.ts:33
    factory index.ts:41
    Context context.ts:69
    _createContext manager.ts:508
    _createOrOpenDocument manager.ts:602
    open manager.ts:378
    openOrReveal manager.ts:417
    execute index.tsx:629
    promise callback*execute index.tsx:629
    execute index.es6.js:352
    values restorablepool.ts:265
    restore restorablepool.ts:255
    restore widgettracker.ts:321
    restore layoutrestorer.ts:259
    activateNotebookHandler index.ts:1505
    promise index.es6.js:1827
    promise callback*activatePlugin index.es6.js:1827
    resolveRequiredService index.es6.js:1914
    required index.es6.js:1822
    activatePlugin index.es6.js:1822
    resolveRequiredService index.es6.js:1914
    required index.es6.js:1822
    activatePlugin index.es6.js:1822
    promises index.es6.js:1990
    start index.es6.js:1989
    main index.out.js:626
    bootstrap bootstrap.js:96
    js bootstrap.js:97
    Webpack 5
index.es6.js:210
```

**Comment by hbcarlos** [comment]
Hi @isms. Thanks for reporting this issue.
That was fixed in #13124, to fix your local env you just need to update the dependency `jupyter_ydoc>=0.2.0`.

**Comment by hbcarlos** [comment]
We will fix this issue in #13168

**Comment by echarles** [comment]
> We will fix this issue in https://github.com/jupyterlab/jupyterlab/pull/13168

Has this been effectively fixed in https://github.com/jupyterlab/jupyterlab/pull/13168?

Question: De we really want to have the outputs history (can be large for binary content like images) in the CRDT structure?

**Comment by hbcarlos** [comment]
Hey @echarles!

Yes, this is now fixed. We can close this issue.

>  De we really want to have the outputs history (can be large for binary content like images) in the CRDT structure?

I think so, at least for now. Indeed, the document's history can get quite big, but we are considering this problem. See: https://github.com/y-crdt/ypy-websocket/pull/53

In addition, for now, every client collaborating on the same document shares the execution environment, so we need to share the results of the execution with everyone.

**Comment by echarles** [comment]
> Yes, this is now fixed. We can close this issue.

@hbcarlos Any link for that?

> I think so, at least for now. Indeed, the document's history can get quite big, but we are considering this problem. 

Are there any figures to illustrate the document growth based on reproducible benchmarking?

**Comment by hbcarlos** [comment]
> @hbcarlos Any link for that?

Yes, it was fixed in #13168

> Are there any figures to illustrate the document growth based on reproducible benchmarking?

No, or at least I'm not aware of that, but it would be great to have some benchmarks.
